src/utils/layers/LayerManager.js

```diff
@@ -84,219 +78,199 @@ export default class LayerManager {
+
+            this._mapLayers[layerConfig.id] = L.tileLayer(tileUrl).addTo(this._map).setZIndex(998);
+
+            this._mapLayers[layerConfig.id].on('load', () => {
+              this._deleteLoader(layerConfig.id);
+            });
+            this._mapLayers[layerConfig.id].on('tileerror', () => {
+              this._deleteLoader(layerConfig.id);
+            });
+          },
+          onError: (data) => {
+            console.error(data);
+            this._deleteLoader(layerConfig.id);
           }
-        };
-
-        this._mapRequests[layerConfig.category] = xmlhttp;
-
+        });
         break;
       }
 
       case 'food': {
+        // Parse config
         const layerConfigConverted = getObjectConversion(layerConfig, options, 'food');
 
-        this._layersLoading[layerConfig.id] = true;
-        const xmlhttp = new XMLHttpRequest();
-        xmlhttp.open('GET', layerConfigConverted.body.url);
-        xmlhttp.send();
-
-        xmlhttp.onreadystatechange = () => {
-          if (xmlhttp.readyState === 4) {
-            if (xmlhttp.status === 200) {
-              const data = JSON.parse(xmlhttp.responseText);
-              const geojson = data.rows[0].data.features || [];
-              this._mapLayers[layerConfig.id] = new BubbleClusterLayer(
-                geojson, layerConfig
-              ).addTo(this._map);
-
-              delete this._layersLoading[layerConfig.id];
-            } else {
-              this._rejectLayersLoading = true;
-            }
-          }
-        };
+        // Save loader
+        this._addLoader(layerConfig.id);
+
+        // Save request && send
+        this._mapRequests[layerConfig.category] = get({
+          url: layerConfigConverted.body.url,
+          onSuccess: (data) => {
+            const geojson = data.rows[0].data.features || [];
+
+            this._mapLayers[layerConfig.id] = new BubbleClusterLayer(
+              geojson, layerConfig
+            ).addTo(this._map);
 
-        this._mapRequests[layerConfig.category] = xmlhttp;
+            this._deleteLoader(layerConfig.id);
+          },
+          onError: (data) => {
+            console.error(data);
+            this._deleteLoader(layerConfig.id);
+          }
+        });
         break;
       }
 
       default: {
-        let layerConfigParsed = {};
-        const layerConfigConverted = getObjectConversion(layerConfig, options, 'water');
-        if (layerConfigConverted.generatedCartocss) {
-          const legendConfigConverted = getObjectConversion(layerSpec.legendConfig, options, 'water');
-          return this._getLegendValues(layerConfigConverted, legendConfigConverted, options);
-        } else {
-          layerConfigParsed = Object.assign({}, layerConfigConverted, Object.assign({
-            body: this._getLayerConfigParsed(layerConfigConverted)
-          }));
+        if (layerConfig.generatedCartocss) {
```

Comment:
Change this and ask for `legendConfig.sqlQuery` param, this way we can get rid of this attribute.